server/app/views/v1/stacks/_stack.json.jbuilder

```diff
@@ -15,5 +15,5 @@ json.variables latest_rev.variables
 json.services stack.grid_services.to_a do |grid_service|
   json.partial! 'app/views/v1/grid_services/grid_service', grid_service: grid_service
 end
-json.parent stack.has_parent? ? { name: stack.parent_name } : nil
-json.children stack.children.pluck(:name).map { |child_name| { name: child_name } }
+json.parent stack.has_parent? ? { name: stack.parent_name, id: "#{stack.grid.try(:name)}/#{stack.parent_name}" } : nil
+json.children stack.children.pluck(:name).map { |child_name| { name: child_name, id: "#{stack.grid.try(:name)}/#{child_name}" } }
```

Comment:
Too much oneliner. Look at how the other structured JSON fields are implemented, these should be something like

```ruby
json.parent do
  if parent_stack = stack.parent_stack
    json.id parent_stack.to_path
    json.name parent_stack.name
  end
end
json.children stack.children do |child_stack|
  json.id child.to_path
  json.name child.name
end
```

Reply:
Seems wrong to create so many extra queries

1) find and load the parent stack to get its name (which is already known at this point)
2) find and load the parent stack grid to get its name (which is already known at this point)
3) find and load the complete child stacks (while we only want the name)
4) find and load the grid for each child to get the same name several times (which is already known at this point)

Reply:
It's an optimization for this specific case, yes... in general, the relations should really be actual mongoid relations, because then `@grid.stacks.includes(:parent_stack, :child_stacks)` should AFAIK be enough to allow writing the straightforward code while avoiding most of the O(N) queries for the `/v1/grids/:grid/stacks` case. The view code would get really ugly if we started to optimize everything like this.

Although it looks like the current `includes(:grid, :grid_services)` isn't enough to optimize out the references in the included documents, as the current code leads to further `{"find"=>"grids", "filter"=>{"_id"=>BSON::ObjectId('580878e5de35781b2c000017')}, "limit"=>1, "singleBatch"=>true}` and `{"find"=>"stacks", "filter"=>{"_id"=>BSON::ObjectId('59c8ffb942132f000ff5fed5')}` queries for each stack. That's presumably for the `stack.grid_services[...].grid/stack` references, and I don't know if mongoid is smart enough to optimize for something like `includes(:grid_services.grid)`?